docs/core/plugins/sieve.md

```diff
@@ -46,128 +58,191 @@ protocol lmtp {
+is currently only applicable for message delivery.
+
+The script from the `discard` storage is only executed when the "implicit keep"
+is canceled, by e.g. the "discard" action, and no actions that deliver the
+message are executed. This "discard script" can prevent discarding the message,
+by executing alternative actions. If the discard script does nothing,
+the message is still discarded as it would be when no discard script is
+configured.
+
+#### `global`
+
+An `global` storage is the source of `:global` include scripts for the Sieve
+include extension. Scripts are accessed by name, so if the storage yields only
+one script, a name must be defined for it; either implicitly by the storage
+driver or explicitly using [[setting,sieve_script_name]]. Multiple `global`
+storages can be configured and each storage will be queried in sequence to
+retrieve the requested script by name. The storages will be queried in the order
+these storages are defined in the configuration until the script is found. The
+order can be overridden by the [[setting,sieve_script_precedence]] setting.
+
+### Script storage drivers
+
+Sieve script storages are implemented as a storage driver. The default
+[file](#sieve_storage_file) storage driver uses the local filesystem. It can use
+a single script file or a directory containing several Sieve script files with a
+symbolic link pointing to the active script.
+
+More complex setups can use other storage drivers such as
+[ldap](#sieve_storage_ldap) or [dict](#sieve_storage_dict) to fetch Sieve
+scripts from LDAP databases or dict storages, respectively.
+
+The storage driver is configured using the [[setting,sieve_script_driver]]
+setting. If not explicitly configured for a [[setting,sieve_script]] block, the
+storage driver is `file` and the default directory is `~/sieve/` with a symbolic
+link at `~/.dovecot.sieve` pointing to the active script file in that directory.
+
+#### Common Settings
+
+All Sieve script storages support the following common settings:
+
+<SettingsComponent tag="sieve-storage" />
 
-### File
+### File storage driver
+
+The `file` script storage driver is used to retrieve Sieve scripts from the file
+system. This is the default type if the `sieve_script_driver` setting is
+omitted.
+
+The path configured using the [[setting,sieve_script_path]] setting can either
+point to a directory or to a regular file. If the path points to a directory, a
+script called `name` is retrieved by reading a file from that directory with the
+file name `name.sieve`.
+
+When a script storage with type `personal` is using the `file` driver and the
+[[setting,sieve_script_path]] points to a directory, a symbolic link points to
+the currently active script (the script executed at delivery). The active script
+can be modified by the user through ManageSieve and by the administrator using
+`doveadm sieve`. The location of this symbolic link can be configured
+using the [[setting,sieve_script_active_path]] setting.
```

Comment:
If sieve_script_path points to a file (not directory), is sieve_script_active_path setting ignored?